### DISCUSSION TITLE
Experiencing a Login Issue

### DISCUSSION THREAD

**Discussion by proj-sashido** [netreconlab/Parse-Swift#59] [Q&A] [answered]
This is my Parse initialisation code: 
```swift
    let url = { … }; let appID = { … }; let clKey = { … }; let liveQueryURL = { … }     
    let credentials = ParseConfiguration(
        applicationId:  appID,
        clientKey:      clKey,
        serverURL:      url,
        liveQueryServerURL: liveQueryURL
    )
    initialize(configuration: credentials)
```

This is working and I’m able to connect to the database, query tables and save objects just fine. 
However when I try to run the following login code:

```swift
    User.login(username: uName, password: pw) { result in
        switch result {
        case .failure(let error): print(error.localizedDescription)
        case .success(let user): /* DO STUFF */ { … }
        }
    }
```

I get the following error:
```swift
ParseError code=-1 error=Error decoding parse-server response: <NSHTTPURLResponse: 0x280c2e4e0> { URL: <redactedURL>/login } { Status Code: 504, Headers {
    "Cache-Control" =     (
        "no-cache"
    );
    Connection =     (
        close
    );
    "Content-Type" =     (
        "text/html"
    );
} } with error: dataCorrupted(Swift.DecodingError.Context(codingPath: [], debugDescription: "The given data was not valid JSON.", underlyingError: Optional(Error Domain=NSCocoaErrorDomain Code=3840 "Invalid value around line 1, column 0." UserInfo={NSDebugDescription=Invalid value around line 1, column 0., NSJSONSerializationErrorIndex=0}))) Format: Optional("<html><body><h1>504 Gateway Time-out</h1>\nThe server didn\'t respond in time.\n</body></html>\n")
```
I know the username/password credentials i’m using to log in are accurate.

### ParseServer
I use a BaaS known as [SashiDo](https://sashido.io), and am currently on ParseServer Version v2.3.3 (I have the option to go up to v3.6.0, but haven’t gotten around to it yet.)
I’ve created a ticket with SashiDo also, in case there’s something on their end causing the issue.

### Questions
- Is there a minimum ParseServer Version that is compatible with ParseSwift?
- Could it have something to do with the fact that `PFUser.logInWithUsername(inBackground:password:)` is a .GET based method, but `User.login(username:password:)` is a .POST request?
- Do you have any other ideas of what the issue might be?

Thank you for any help you can provide!

**Comment by cbaker6** [comment, marked as answer]
Hi @project-academy, I discuss using older servers here: https://github.com/netreconlab/Parse-Swift/issues/29#issuecomment-1344577760

> Is there a minimum ParseServer Version that is compatible with ParseSwift?

Yes, see here: https://community.parseplatform.org/t/parse-swift-minimum-parse-server-version/2419/4?u=cbaker6

> Could it have something to do with the fact that PFUser.logInWithUsername(inBackground:password:) is a .GET based method, but User.login(username:password:) is a .POST request?

See issues with GET: https://github.com/parse-community/Parse-SDK-JS/issues/1283#issue-782120829

I recommend using a service that allows you to take advantage of the latest servers (Which is currently on Parse Server 6) or self host. I have a self hosting example using Heroku here: https://github.com/netreconlab/parse-hipaa/tree/snapcat#deployment

**Reply by proj-sashido** [reply]
Ah yep, I personally tested this on later versions (as low as v2.8.4) and there’s no issues with login. Using .POST to login was added in v2.6.4 according to the release logs.

Unfortunately our organisation doesn't have the capacity/ability to self host but we’ll be looking to make sure the BaaS we go with offers up-to-date Parse Servers.

Thank you for being patient with my adoption of ParseSwift!
